### PR TITLE
Add error check in kubectl proxy on server setup

### PR DESCRIPTION
**Pull request by odinuge**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
/kind bug

**What this PR does / why we need it**:
If "NewServer" returns an error, it will result in a nil pointer
dereference segfault.

A simple way to test the behavior is to prefix the server url with a
colon, ":".

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

I guess I can wrap the error in a `error when starting server` if that is preferred over this (I just used the same as the other error handler in the file). 
/sig cli

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by aojea** [comment]
actually it seems to recover the previous behavior
https://github.com/kubernetes/kubernetes/blame/cd30bd6167245a8e6dc623575398e9618359c9cd/pkg/kubectl/cmd/proxy.go

@odinuge are you still working on this? are you going to rebase this PR?

**Comment by odinuge** [comment]
Rebased the PR now @aojea :smile: 

/remove-lifecycle rotten
/priority backlog

**Comment by aojea** [comment]
@liggitt is ok to LGTM?

File: staging/src/k8s.io/kubectl/pkg/cmd/proxy/proxy.go [hunk 1]
```diff
@@ -200,6 +200,10 @@ func (o ProxyOptions) Validate() error {
 func (o ProxyOptions) RunProxy() error {
 	server, err := proxy.NewServer(o.staticDir, o.apiPrefix, o.staticPrefix, o.filter, o.clientConfig, o.keepalive)
 
+	if err != nil {
+		return err
```
Comment (liggitt) [inline staging/src/k8s.io/kubectl/pkg/cmd/proxy/proxy.go]: this looks correct. the other `klog.Fatal(err)` lines in this method should be changed to return the err instead as well

**Comment by aojea** [comment]
/retest

failed 

[k8s.io] Docker Containers should be able to override the image's default command and arguments [NodeConformance] [Conformance] expand_less

**Pull request merged**